code/__DEFINES/components.dm

```diff
@@ -213,6 +213,23 @@
 #define COMSIG_OBJ_SETANCHORED "obj_setanchored"				//called in /obj/structure/setAnchored(): (value)
 #define COMSIG_OBJ_DEFAULT_UNFASTEN_WRENCH "obj_default_unfasten_wrench"	//from base of code/game/machinery
 #define COMSIG_OBJ_UPDATE_ICON	"obj_update_icon"				//Called by obj to itself whenever update_icon is run. Useful for containers that want to detect this change
+#define COMSIG_OBJ_ATTACK_ANIMAL "obj_attack_animal"			//from base obj/attack_animal()
+
+// /obj/machinery signals
+#define COMSIG_MACHINERY_POWERRESTORED "machinery_powerrestored"	//from base machinery/power_change()
+#define COMSIG_MACHINERY_POWERLOSS "machinery_powerloss"
+#define COMSIG_MACHINERY_BROKEN "machinery_broken"
+
+#define COMSIG_AIRLOCK_BUMPOPEN "airlock_bumpopen"					//from airlock bumpopen()
+	#define COMSIG_ELECTRIFIED_SHOCK 1

```

Comment:
what is going on

Reply:
\shrug

Reply:
Why is it a number

Reply:
Nevermind I see

Reply:
It's a bitflag return value for the above signal, this is the normal format in this file